tests/TestCadQuery.py

```diff
@@ -362,6 +362,32 @@ def testRevolveCone(self):
         self.assertEqual(2, result.faces().size())
         self.assertEqual(2, result.vertices().size())
         self.assertEqual(3, result.edges().size())
+        
+    def testSpline(self):
+        """
+        Tests construction of splines
+        """
+        pts = [
+            (0, 1),
+            (1, 2),
+            (2, 4)
+        ]
+
+        # Spline path - just a smoke test
+        path = Workplane("XZ").spline(pts).val()
+
+        # Closed spline
+        path_closed = Workplane("XZ").spline(pts,periodic=True).val()
+        self.assertTrue(path_closed.IsClosed())
+        
+        # attempt to build a face
+        w = Wire.assembleEdges([path_closed,])
+        f = Face.makeFromWires(w)
```

Comment:
Do you want to add an assert here to make sure that the face was built correctly?

Reply:
Yes, good catch! Actually the `isValid` method was not implemented. Now I check if the generated face is valid.